### ISSUE TITLE
false positive LabeledExpression in conjunction with this@ClassName.

### ISSUE THREAD

**Issue by robstoll** [detekt/detekt#1226]
## Expected Behavior
Maybe some people want to have a report for `this@ClassName...` as well. I for one don't see the reason for it and all cases I had were not fixable unless I introduce a local variable with a different name but that is even uglier. I appreciate other hints such as `return@...` though, so I don't want to disable the rule all together. 
So... depending on whether there are good use cases for `this@ClassName...` it could be a configuration flag something like `ignoreThisLabel` or the rule ignores such labels.

## Current Behavior
They get reported

## Your Environment

* Version used: 1.0.0.RC8
* Link to your project: https://github.com/loewenfels/dep-graph-releaser/blob/master/dep-graph-releaser-gui/src/main/kotlin/ch.loewenfels.depgraph/gui/components/Pipeline.kt#L117

**Comment by schalkms** [comment]
The `this@modifier` is redundant in your code, isn't it?

**Comment by robstoll** [comment]
Not sure what you mean. Could you point to a line number?

**Comment by schalkms** [comment]
I meant the link to `Pipeline.kt#117` that you have included in your initial post.

**Comment by schalkms** [comment]
I see. It's referencing a static method called `contextMenu` in this context.
I think there should be a list of ignored labels as parameter and not just ignoring this specific use case here. The user has the option to configure it. For instance
`ignoredLabels = "this,return,break.."`

**Comment by robstoll** [comment]
`contextMenu` is not a static method. It is a private field, hence `this` is necessary.
That's the reason why IMO it is a false positive, you cannot leave out `this@Pipeline`. Or in other words, IMO no ignore should be required in such situations.

**Comment by schalkms** [comment]
In your context you are referencing the member variable of the `Pipeline` class. If you remove the `this@Pipeline` from your code you are referencing a static method.

**Comment by robstoll** [comment]
Exactly, not the same behaviour thus you cannot remove `this`... ah sorry, I mixed up the issue (reported too many lately). It's not about an unnecessary label but about the usage of labels as such. I think the ignoreLabel makes sense :+1:

**Comment by schalkms** [comment]
Another false positive

```Kotlin
class Foo {
  fun Bar.extensionMethod() {
    add(this@Foo)
  }
}
```
